students/WesleyM/Lesson10/mailroom_oo.py

```diff
@@ -0,0 +1,172 @@
+from functools import reduce
+
+class Donor:
+    def __init__(self, name, donation=None):
+        self.donation = donation if donation is not None else []
+        self.name = name
+    @property
+    def total(self): 
+        return sum(self.donation)
+
+    @property
+    def average(self):
+        if donation:
+            return self.total/len(self.donation)
+        else:
+            return 0
+    
+    def add_donation(self, amount):
+        self.donation.append(amount)
+        self.total += amount
+        self.average = self.total / len(self.donation)
+    
+    def factor_donation(self, factor, min_donation, max_donation):
+        return list(map(lambda x: x * factor,
+                        self.filter_donations(self.donation,
+                                              min_donation, max_donation)))
+    
+    @staticmethod
+    def filter_donations(donations, min_donation, max_donation):
+        if max_donation is None:
+            return list(filter(lambda x: x >= min_donation, donations))
+        else:
+            return list(filter(lambda x: min_donation <= x <= max_donation,
+                               donations))
+
+class DonorDatabase:
+
+    def __init__(self, name=None):
+        self.donors = [] if name is None else name
+    
+    def add_donor(self, name):
+        self.donors.append(name)
+        
+    def get_donor(self, name):
+        for d in self.donors:
+            if d.name == name:
+                return d
+
+    def get_all_donor_names(self):
+        return [donor.name for donor in self.donors]
+        
+    def sort_donors(self):
+        return sorted(self.get_all_donor_names())
+    
+    def donor_input(self):
+        return input("Enter a donor name or input 'List' for a list of donors\n>")
+    
+    def send_thankyou(self):
+        don_input = None
+        while not don_input:
+            don_input = self.donor_input()
+            if don_input.lower() == "list":
+                print(self.sort_donors())
+                don_input = None
+
+        donation = None
+        while not donation:
+            try:
+                donation = self.donation_prompt()
+            except ValueError:
+                print("Enter donations numerically")
+
+        if don_input in self.get_all_donor_names():
+            for d in self.donors:
+                if d.name == don_input:
+                    d.add_donation(donation)
+        else:
+            try:
+                d2 = Donor(don_input, [donation])
+                self.add_donor(d2)
+            except ValueError:
+                print("Please enter a name")
+
+        print("Thank you {} for your donation of ${:.2f}"
+              .format(don_input, donation))
+    
+    def donation_prompt(self):
+        return float(input("Enter a donation amount \n>"))
```

Comment:
you did a really good job separating user interaction code from your DonorDatabase class; This should probably be outside of this class as well.